client/src/components/QandA/QuestionsList.jsx

```diff
@@ -1,45 +1,122 @@
+import { token, url } from '/config.js';
 import QuestionCard from './QuestionsList/QuestionCard.jsx';
 import AddQuestionOrAnswer from './QuestionsList/QuestionCard/AddQuestionOrAnswer.jsx';
 
-// const host = 'https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp';
-const headers = {
-  'Authorization': token
-};
+// const headers = {
+//   'Authorization': token
+// };
 
-const QuestionsList = (props) => {
-  const [questions, setQuestions] = useState([]);
+class QuestionsList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      questions: [],
+      questionCounter: 0
+    };
+    this.onMoreQuestionsClick = this.onMoreQuestionsClick.bind(this);
+    this.onCollapseQuestionsClick = this.onCollapseQuestionsClick.bind(this);
+  }
 
-  useEffect(() => {
-    axios.get(url + `/qa/questions?product_id=${props.product_id}`, { headers })
+  componentDidMount() {
+    axios.get('http://127.0.0.1:3000/qa/questions', {params: { product_id: this.props.product_id }})
       .then(returnedQuestions => {
-        setQuestions(returnedQuestions.data.results);
+        this.setState({
+          questions: returnedQuestions.data.results,
+          questionCounter: 2
+        });
       })
-      .catch(err => console.error('Get request error (QuestionsList.jsx): ', err))
-  }, []);
-
-  return (
-    <>
-      {questions.map(item =>
-        <QuestionCard
-          answers={item.answers}
-          asker={item.asker_name}
-          body={item.question_body}
-          date={item.question_date}
-          helpful={item.question_helpfulness}
-          key={item.question_id}
-          reported={item.reported}
-          ID={item.question_id}
+      .catch(err => console.error('Error receiving response (QuestionsList.jsx): ', err));
+  }
+
+  onMoreQuestionsClick() {
+    if (this.state.questions.length - this.state.questionCounter <= 2) {
+      this.setState({
+        questionCounter: this.state.questions.length
+      });
+    } else {
+      this.setState({
+        questionCounter: this.state.questionCounter + 2
+      });
+    }
+  }
+
+  onCollapseQuestionsClick() {
+    this.setState({
+      questionCounter: 2
+    });
+  }
+
+  render() {
+    let sortedQuestions = this.state.questions.sort((a, b) =>
+      b.question_helpfulness - a.question_helpfulness);
+
+    return (
+      <>
+        {sortedQuestions.slice(0, this.state.questionCounter).map(item =>
+          <QuestionCard
+            answers={item.answers}
+            asker={item.asker_name}
+            body={item.question_body}
+            date={item.question_date}
+            helpful={item.question_helpfulness}
+            key={item.question_id}
+            reported={item.reported}
+            question_id={item.question_id}
+            product_name={this.props.product_name}
+          />
+        )}
+        {this.state.questions.length <= 2
+        // || this.state.questions.length === this.state.questionCounter)
```

Comment:
what is this logic for? saving got later function?

Reply:
If you are asking about the commented out line, it was something I thought I might be coming back to. Seems unlikely now. Will remove. I might have a few of these lying around. Will need to do some thorough scrubbing.